### PR TITLE
[INFINITY-1074] Authorize marathon to launch tasks as root in strict mode tests

### PR DESCRIPTION
**Pull request by mgummelt**
I've also refactored `setup_permissions.sh` to be organized by permission, and I've changed the comments in that file, which were incorrect.

### CODE REVIEW

**Comment by mgummelt** [comment]
retest this please

**Comment by susanxhuynh** [comment]
@mgummelt dcos-cassandra-service/test.sh and dcos-kafka-service/test.sh both call setup_permissions.sh. They would have to be updated, too.

**Comment by mgummelt** [comment]
@susanxhuynh good catch.  I'll update those.

**Comment by mgummelt** [comment]
That shouldn't block this PR, though, since this has to go in first.

**Comment by mgummelt** [comment]
I just removed the interface change to `setup_permissions.sh`

**Comment by mgummelt** [comment]
cc @susanxhuynh for approval.

File: tools/setup_permissions.sh [hunk 1]
```diff
@@ -19,81 +20,94 @@ else
...
+         -d '{"description":"Execute Mesos tasks as user=${LINUX_USER}"}' \
+         -H 'Content-Type: application/json'
+    curl -k -L -X PUT \
+         -H "Authorization: token=${ACS_TOKEN}" \
+         "${DCOS_URL}/acs/api/v1/acls/dcos:mesos:master:task:user:${LINUX_USER}/users/${SERVICE_ACCOUNT_NAME}/create"
+
+    # In order for the Spark Dispatcher to register with Mesos as
+    # root, we must launch the dispatcher task as root.  The other
+    # frameworks are launched as nobody, but then register as
+    # service.user, which defaults to root
```
Comment (susanxhuynh) [inline tools/setup_permissions.sh]: I'm not following this. Why can't Spark Dispatcher do the same as the other frameworks? Launch as nobody and then register as root?

Comment (mgummelt) [inline tools/setup_permissions.sh]: The dispatcher registers as the current linux user.  It's not configurable.  We could make it configurable, but there's really no reason to.

Comment (susanxhuynh) [inline tools/setup_permissions.sh]: OK

**Pull request merged**